src/sentry/buffer/base.py

```diff
@@ -189,7 +197,42 @@ def process(
                     # continue
                     pass
                 else:
-                    group.update(using=None, **update_kwargs)
+                    try:
+                        group.update(using=None, **update_kwargs)
+                    except DataError as e:
+                        # Catch NumericValueOutOfRange when times_seen exceeds 32-bit limit
+                        # This happens when current_times_seen + increment > MAX_INT32
+                        if (
+                            isinstance(e.__cause__, psycopg2.errors.NumericValueOutOfRange)
+                            and "times_seen" in columns
+                        ):
+                            # Cap times_seen to MAX_INT32 and retry the update
+                            update_kwargs["times_seen"] = MAX_INT32
+                            try:
+                                group.update(using=None, **update_kwargs)
+                                logger.warning(
```

Comment:
Could be worth logging this to datadog instead. I think that warnings will end up as sentry errors too, so you might want to drop these to info.